pkg/identity/gitlabcom/principal_test.go

```diff
@@ -111,6 +118,45 @@ func TestJobPrincipalFromIDToken(t *testing.T) {
 			WantErr:     true,
 			ErrContains: "project_path",
 		},
+		`Token missing pipeline_sha claim is ok`: {
```

Comment:
```suggestion
		`Token missing ci_config_sha claim is ok`: {
```

Worth adding a test case for missing `ci_config_ref_uri` that should fail?

Reply:
Good call, thanks!